simulation_ws/src/test_nodes/nodes/docking_test.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+
+import rospy
+import rostest
+import time
+import os
+import unittest
+import tf
+
+from math import hypot
+from std_msgs.msg import Bool
+from geometry_msgs.msg import PoseStamped
+from rosgraph_msgs.msg import Clock
+from nav_msgs.msg import Odometry
+from std_srvs.srv import Trigger
+
+from robomaker_simulation_msgs.msg import Tag
+from robomaker_simulation_msgs.srv import Cancel, AddTags
+
+class DockingTest(unittest.TestCase):
+
+    def runTest(self):
+        # Start the docking test
+        self.test_docking()
+
+    def setUp(self):
+        self.test_name = "Docking_Test_" + \
+            str(self.get_time())
+        rospy.loginfo("Test Name: %s", self.test_name)
+        self.is_docked = False
+        self.is_docking = False
+        self.is_cancelled = False
+        self.get_params()
+        while rospy.Time.now().secs == 0:
+            pass
+        self.sim_start_time_secs = rospy.Time.now().secs
+        self.timer = rospy.Timer(self.sec_before_docking, callback=self.trigger_docking_client(), oneshot=True)
+
+    def trigger_docking_client(self):
+        rospy.wait_for_service('trigger_docking_service')
+        try:
+            client_trigger = rospy.ServiceProxy('trigger_docking_service', Trigger)
+            self.set_tag(name="Docking_process_start",
+                         value=str(self.get_time()) )
+            response = client_trigger()
+            self.is_docking = True
+            self.dock_start_time_secs = rospy.Time.now().secs
+            return response.success
+        except rospy.ServiceException as e:
+            self.set_tag(name=self.test_name + "Failed", value=str(time.time()).split(".",1)[0])
+            self.cancel_job()
+            rospy.logerr("Service call failed: %s"%e)
+
+    def test_docking(self):
+        try:
+            rospy.loginfo(self.test_name)
+            self.set_tag(name="Time_Elapsed_Start",
+                         value=str(self.sim_start_time_secs))
+            if self.docked_topic_enabled is True:
+                rospy.loginfo("%s topic listening enabled.", self.docked_topic_name)
+                rospy.Subscriber(self.docked_topic_name, Bool, self.is_docked_cb)
+            elif self.odom_topic_enabled is True:
+                rospy.loginfo("%s topic listening enabled.", self.odom_topic_name)
+                rospy.Subscriber(self.odom_topic_name, Odometry, self.odom_cb)
+            else:
+                rospy.logerr("Neither %s or %s topics are enabled", self.odom_topic_name, self.docked_topic_name)
+                self.set_tag(name="Status", value="Failed")
+                self.set_tag(name="No_topics_enabled", value=str(self.get_time()))
```

Comment:
I don't really like the format, but for consistency, all UpperCase separated with an underscore.